ktor-server/ktor-server-plugins/ktor-server-sse/jvmAndNix/src/io/ktor/server/sse/DefaultServerSSESession.kt

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2014-2023 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package io.ktor.server.sse
+
+import io.ktor.server.application.*
+import io.ktor.sse.*
+import io.ktor.util.*
+import io.ktor.utils.io.*
+import kotlinx.coroutines.sync.*
+import kotlin.coroutines.*
+
+internal class DefaultServerSSESession(
+    private val output: ByteWriteChannel,
+    override val call: ApplicationCall,
+    override val coroutineContext: CoroutineContext
+) : ServerSSESession {
+    private val mutex = Mutex()
```

Comment:
why is it needed?

Reply:
Because concurrent writing is not supported for `ByteWriteChannel` and we need to order writing events